cosmwasm/submit-proposal.js

```diff
@@ -278,15 +278,18 @@ const instantiateChainContracts = async (client, config, options, _args, fee) =>
         throw new Error('Salt is required when instantiating chain contracts');
     }
 
+    const chainConfig = config.getChainConfig(chainName);
+    const multisigProverContractName = getProverContractForChainType(chainConfig.chainType);
+
     // validate that the contract configs exist
     let gatewayConfig = config.getContractConfigByChain(GATEWAY_CONTRACT_NAME, chainName);
     let verifierConfig = config.getContractConfigByChain(VERIFIER_CONTRACT_NAME, chainName);
-    let proverConfig = config.getContractConfigByChain(MULTISIG_PROVER_CONTRACT_NAME, chainName);
+    let proverConfig = config.getContractConfigByChain(multisigProverContractName, chainName);
```

Comment:
yeah like we discussed ideally this could just be something like

```suggestion
    let gatewayConfig = config.getGatewayConfig(chainName);
    let verifierConfig = config.geVerifierConfig(chainName);
    let proverConfig = config.getProverConfig(chainName);
```

or better yet if the config has already been retrieved for the context of the chain
```
config.gateway(chain)
```
can replace
```
let gatewayConfig ...;
```
entirely